### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by python4eg**

### CODE REVIEW

File: static/js/pages/UserPage/UserPage.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import Result from './Result';
+
+class User extends React.Component {
+   constructor() {
+        super();
+        this.state = { user: [] };
+    }
+
+    componentDidMount() {
+        fetch(`http://localhost:3090/api/profile`)
```
Comment (python4eg) [inline static/js/pages/UserPage/UserPage.js]: Should be configurable

File: static/js/pages/UserPage/UserPage.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            .then(response => response.json() )
```
Comment (python4eg) [inline static/js/pages/UserPage/UserPage.js]: error handling for json parding

File: static/js/pages/UserPage/Result.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            <div className="container">
+                {profile}
+            </div>
+        );
+    }
+}
+
+class ResultItem extends React.Component{
+    render(){
+        var user = this.props.user;
```
Comment (python4eg) [inline static/js/pages/UserPage/Result.js]: PropTypes

File: static/js/pages/UserPage/Result.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        user.is_premium == false ? user.is_premium = "none" : user.is_premium = "active";
```
Comment (python4eg) [inline static/js/pages/UserPage/Result.js]: why?

File: static/js/pages/UserPage/Result.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    color: '#357786',
+    padding: '20px 0'
+}
+
+const nameStyle = {
+    color: '#357786',
+    padding: '20px',
+    textAlign: 'center'
+}
+
```
Comment (python4eg) [inline static/js/pages/UserPage/Result.js]: too many spaces

File: static/js/pages/RecipesPage/RecipeItem.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router';
+import { Grid, Row, Col } from 'react-bootstrap';
+
+const imgStyle = {
```
Comment (python4eg) [inline static/js/pages/RecipesPage/RecipeItem.js]: why don't you put it into css?

File: static/js/pages/SignInPage/SignInForm.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    handleSubmit(e) {
+        e.preventDefault();
+
+        let credentials = {};
+        credentials.email = this.state.email;
+        credentials.password = this.state.password;
+
+        axios.post('/api/login', credentials)
```
Comment (python4eg) [inline static/js/pages/SignInPage/SignInForm.js]: https://en.wikipedia.org/wiki/Basic_access_authentication

File: controllers/recipeController.js [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+    for (var i = 0; i < arrayTags.length; i++) {
+        arrayTags[i] = arrayTags[i].replace(/[-+().!@#$%^&' "*<>\s]/g, '');
+    }
+    return arrayTags;
+}
+
+recipeController.createRecipe = (req, res, next) => {
+    let form = new multiparty.Form();
+
+    form.parse(req, (err, fields, files) => {
```
Comment (python4g) [inline controllers/recipeController.js]: http://webuniverse.io/cyclomatic-complexity-refactoring-tips/

File: controllers/recipeController.js [hunk 9]
```diff
@@ -0,0 +1,359 @@
...
+        if (!status) {
+            res.sendStatus(500)
+        }
+        recipeData.addValue.forEach((value) => {
+            db.query(recipeObject.updateRecipe(recipeData, value.name), (error, result) => {
+                if (result.rows) {
+                    const insertId = result.rows[0].id;
+                    db.query(recipeObject.addDbLink(table, fieldConnect, recipeData.id, insertId), (error) => {
+                        if (error) {
+                            status = false;
```
Comment (python4g) [inline controllers/recipeController.js]: assign status number here

File: controllers/recipeController.js [hunk 10]
```diff
@@ -0,0 +1,359 @@
...
+        });
+        if (status) {
+            res.sendStatus(200)
+        } else {
+            res.sendStatus(500);
+        }
+    } else {
+        db.query(recipeObject.upsertData(recipeData), (err) => {
+            if (err) {
+                res.sendStatus(500);
```
Comment (python4g) [inline controllers/recipeController.js]: assign to `status` var

File: controllers/recipeController.js [hunk 11]
```diff
@@ -0,0 +1,359 @@
...
+            } else {
+                res.sendStatus(200);
+            }
+        });
```
Comment (python4g) [inline controllers/recipeController.js]: send status outside `if..else` statement

File: db/config.example.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
+var userName = "userName";
```
Comment (python4g) [inline db/config.example.js]: lowercase

File: index.html [hunk 13]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/latest/css/bootstrap.min.css" />
```
Comment (python4g) [inline index.html]: `type` is missed

File: static/js/common/EditableImage.js [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.status === 'success') {
+            this.setState({
+                message: 'Saved',
+                saving: false,
+                editing: false });
+            wait(3000)
+            .then(() => {
```
Comment (python4g) [inline static/js/common/EditableImage.js:60]: 👍

File: static/js/common/EditableList.js [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+        super(props);
+        this.state = {
+            cancelItem: [],
+            deletedItem: [],
+            items: this.props.data,
+            editing: false,
+            saving: false,
+            message: '',
+        };
+        this.edit = this.edit.bind(this);
```
Comment (python4g) [inline static/js/common/EditableList.js:46]: what about es6 way??
```javascript
 this.edit = () => this.edit());
```

File: static/js/common/EditableList.js [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+        onSave: null,
+        items: [],
+    }
+
+    static propTypes = {
+        fieldName: PropTypes.string,
+        onSave: PropTypes.func,
+    }
+
+    constructor(props) {
```
Comment (python4g) [inline static/js/common/EditableList.js:36]: what about es6 way with class properties?
```javascript
edit = () => this.edit();
saveEdit = () => this.saveEdit();
cancelEdit = () =>  this.cancelEdit();
constructor(props) {
        super(props);
        this.state = {
            cancelItem: [],
            deletedItem: [],
            items: this.props.data,
            editing: false,
            saving: false,
            message: '',
        };
}
```

File: static/js/common/EditableList.js [hunk 17]
```diff
@@ -0,0 +1,215 @@
...
+                            disabled = { this.state.saving } >
+                        Cancel 
+                        </button> 
+                        { this.renderSavingAnimation() } 
+                    </div>
+                </div>
+                );
+            }
+            return ( 
+                <div>
```
Comment (python4g) [inline static/js/common/EditableList.js]: nitpick: Inheritance indent for readability 
```javascript
<div>
    <div>
       <ul className = "list-group" > 
           { items } 
       </ul>
   </div>
</div>
```

File: static/js/common/EditableList.js [hunk 18]
```diff
@@ -0,0 +1,215 @@
...
+                );
+            }
+        };
+
+
+        class AddItemForm extends Component {
+            createItem(e) {
+                e.preventDefault();
+                var item = this.refs.itemName.value;
+                if (typeof item === 'string' && item.length > 0) {
```
Comment (python4g) [inline static/js/common/EditableList.js]: https://lodash.com/docs#isString and https://lodash.com/docs/4.17.4#isEmpty

File: static/js/common/EditableText.js [hunk 19]
```diff
@@ -0,0 +1,182 @@
...
+            });
+        } else {
+            this.props.onSave(this.props.updateId, this.props.fieldName, this.input.value);
+            this.setState({text: this.input.value});
+        }
+    }
+
+    renderSavingAnimation() {
+        if (this.state.saving) {
+            return <ReactLoading style={centerDiv} type="bars" color="#444" height="70" width="20" />;
```
Comment (python4g) [inline static/js/common/EditableText.js:116]: move repeatable values into constants.
I've seen `bars` and color `#444` many times

File: static/js/common/Header.js [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+            })
+            .catch(err => {
+                console.log(err.stack);
+                console.log('Failed to log out');
+            });
+    }
+
+    componentWillMount() {
+        if (document.cookie) {
+            this.setState({ loggedIn: true });
```
Comment (python4g) [inline static/js/common/Header.js]: `this.setState({ loggedIn: document.cookie});`

File: static/js/layouts/MainLayout.js [hunk 21]
```diff
@@ -0,0 +1,16 @@
+import React, { Component } from 'react';
+
+import Footer from './Footer';
+
+class MainLayout extends Component {
+    render() {
+        return (
+            <div>
+                    {this.props.children}
```
Comment (python4g) [inline static/js/layouts/MainLayout.js]: indent

File: static/js/common/EditableList.js [hunk 22]
```diff
@@ -0,0 +1,215 @@
...
+            color = "#444"
+            height = "70"
+            width = "20" / > ;
+        }
+        return null;
+    }
+
+
+    render() {
+        const items = this.state.items.map(function(item, index) {
```
Comment (python4g) [inline static/js/common/EditableList.js]: arrow function

File: static/js/pages/CreateRecipePage/CreateRecipePage.js [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+        theme: 'light',
+        time: 4000,
+        transition: 'fade',
+        type: 'success',
+    }
+
+    addRecipe(recipe) {
+        return new Promise((resolve, reject) => {
+            const data = new FormData();
+            data.append('title', recipe.title);
```
Comment (python4g) [inline static/js/pages/CreateRecipePage/CreateRecipePage.js]: why not just:
```
for ( var key in recipe) {
    data.append(key, recipe[key]);
}
```

File: static/js/pages/CreateRecipePage/CreateRecipePage.js [hunk 24]
```diff
@@ -0,0 +1,63 @@
...
+        offset: 14,
+        position: 'top right',
+        theme: 'light',
+        time: 4000,
+        transition: 'fade',
+        type: 'success',
+    }
+
+    addRecipe(recipe) {
+        return new Promise((resolve, reject) => {
```
Comment (python4g) [inline static/js/pages/CreateRecipePage/CreateRecipePage.js]: 👍  for `Promise` and what about using `axios`?

File: static/js/common/EditableText.js (hunk 19 above)
Comment (StasMelnyk) [reply static/js/common/EditableText.js:116]: This is temporary for now. In the future, we will create new preloader

File: config.js [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+
+export const nodeEnv = env.NODE_ENV || 'development';
+
+export const logStars = (message) => {
+    console.info('********');
+    console.info(message);
+    console.info('********');
+};
+
+export default {
```
Comment (python4eg) [inline config.js]: Do You still use it?

**Pull request closed without merging**